share/emock.sh

```diff
@@ -142,26 +147,23 @@ emock()
         # Create state directory
         statedir="'${statedir}'"
 
-        # Update call count
-        called=0
-        if [[ -e "${statedir}/called" ]]; then
-            called=$(cat "${statedir}/called")
-            (( called++ ))
-        fi
-        echo ${called} > "${statedir}/called"
+        # Get current call count for all our state files. Then update called count inside the file.
+        # Do not modify our local variable as that would cause us to write out to the wrong state files.
+        called=$(cat "${statedir}/called")
+        echo "$(( called + 1 ))" > "${statedir}/called"
 
         # Create directory to store files in for this invocation
         mkdir -p "${statedir}/${called}"
 
         # Save off timestamp and argument array
         echo -en $(date "+%FT%TZ") > "${statedir}/${called}/timestamp"
-        printf "\"%s\" " "${@}" > "${statedir}/${called}/args"
+        printf "%s\n" "${@}" > "${statedir}/${called}/args"
```

Comment:
Make the arg file be newline delimited. Otherwise, there's no way to safely preserve whitespace in the file and get it back out in any sensible way. With ebash we can use `array_init_nl` to parse this back. This was wrong all along, but the errors were being f'ing suppressed b/c I'm dumb.